src/digitalmarketplace/components/list-input/list-input.js

```diff
@@ -0,0 +1,193 @@
+import { nodeListForEach } from '../../common'
+
+var getSibling = function (direction, elem, selector) {
+  // Get the next sibling element
+  var sibling = (direction === 'next') ? elem.nextElementSibling : elem.previousElementSibling
+
+  // If there's no selector, return the first sibling
+  if (!selector) return sibling
+
+  // If the sibling matches our selector, use it
+  // If not, jump to the next sibling and continue the loop
+
+  // For IE10,11 (Polyfill for matches)
+  if (!Element.prototype.matches) {
+    Element.prototype.matches = Element.prototype.msMatchesSelector ||
+                                Element.prototype.webkitMatchesSelector
+  }
+  while (sibling) {
+    if (sibling.matches(selector)) return sibling
+    sibling = (direction === 'next') ? sibling.nextElementSibling : sibling = sibling.previousElementSibling
+  }
+}
+
+function ListInput ($module) {
+  this.$module = $module
+  this.$allVisibleItems = $module.querySelectorAll('.dm-list-input__item:not(.dm-list-input__item--hidden)')
+  this.$addAnotherButton = $module.querySelector('.dmp-list-input__item-add')
+  this.$allItems = $module.querySelectorAll('.dm-list-input__item')
+  this.visibleItems = 0
+}
+
+ListInput.prototype.init = function () {
+  // Check for module
+  if (!this.$module) {
+    return
+  }
+
+  this.hideEmptyItems()
+  this.updateAllCounters()
+
+  this.bindRemoveClickEvent()
+
+  this.toggleAddAnotherButton()
+  this.bindAddClickEvent()
+}
+
+// Hide all items that do not have a value (except for the first one, or the first two if no items have a value)
+ListInput.prototype.hideEmptyItems = function () {
+  var numberOfVisibleEmptyItems = 0
+  var numberOfFilledInItems = 0
+  nodeListForEach(this.$allItems, function ($item, index) {
+    var $input = $item.querySelector('.dm-list-input__item-input')
+    var $removeButton = $item.querySelector('.dm-list-input__item-remove')
+
+    if ($input.value === '') {
+      if (numberOfVisibleEmptyItems === 0) {
+        $removeButton.classList.remove('dm-list-input__item-remove--hidden')
+      } else if (numberOfVisibleEmptyItems === 1 && numberOfFilledInItems === 0) {
+        $removeButton.classList.remove('dm-list-input__item-remove--hidden')
+      } else {
+        $item.classList.add('dm-list-input__item--hidden')
+        $removeButton.classList.add('dm-list-input__item-remove--hidden')
+        $input.setAttribute('disabled', 'true')
+      }
+      numberOfVisibleEmptyItems += 1
+    } else {
+      $item.classList.remove('dm-list-input__item--hidden')
+      $removeButton.classList.remove('dm-list-input__item-remove--hidden')
+      numberOfFilledInItems += 1
+    }
+  })
+  this.$allVisibleItems = this.$module.querySelectorAll('.dm-list-input__item:not(.dm-list-input__item--hidden)')
```

Comment:
seems like this `:not` selector is not supported in IE? https://caniuse.com/#search=%3Anot Do we need to find an alternative way?

Reply:
Oof, good catch